catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ResultIterable.java

```diff
@@ -190,4 +248,26 @@ private void copyQueryRequestAndQuery(QueryRequest queryRequest) {
                     queryRequest.getProperties());
         }
     }
+
+    private static class ResultListIterator implements Iterator<List<Result>> {
+        private final Iterator<Result> iterator;
+
+        private final int pageSize;
+
+        private ResultListIterator(Iterator<Result> iterator, int pageSize) {
+            this.iterator = iterator;
+            this.pageSize = pageSize;
+        }
+
+        @Override
+        public boolean hasNext() {
+            return iterator.hasNext();
+        }
+
+        @Override
+        public List<Result> next() {
+            return stream(iterator).limit(pageSize)
```

Comment:
❗️ Need to double check that streaming over the iterator will not consume more than the `limit` in any scenario. And streaming multiple times over the same iterator causes no ill effects.